lib/publiccloud/azure.pm

```diff
@@ -61,65 +72,111 @@ sub get_image_id {
     return $self->SUPER::get_image_id($img_url);
 }
 
-=head2 generate_azure_image_definition
+=head2 find_blob
 
-    my $definition = $self->generate_azure_image_definition();
+Finds the image blob in Azure
 
-Generated the Azure Image name from the job settings. If present, it takes the PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION setting.
-If not present it generated the image definition name based on distri, version, flavor and SKU.
-
-Note: Image definitions needs to be distinct names and can only serve one architecture!
+ * Requires image name parameter, e.g. 'SLES15-SP5-BYOS.aarch64-1.0.0-Azure-Build1.68.xz'
+ * Returns 0 if the image blob  has not been found or 1 if the image blob has ben found.
 
-Example: 'SLE-MICRO-5.4-BYOS-AZURE-X86_64-GEN2'
 =cut
 
-sub generate_azure_image_definition {
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
-    my $definition = get_required_var('DISTRI') . '-' .
-      get_required_var('VERSION') . '-' .
-      get_required_var('FLAVOR') . '-' .
-      $arch . '-' .
-      get_var('PUBLIC_CLOUD_AZURE_SKU', $default_sku);
-    return get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
-}
-
-sub find_img {
+sub find_blob {
```

Comment:
```suggestion
sub img_blob_exists {
```
Nitpicking here, but I think a subroutine `img_blob_exists` is more descriptive. `find_blob` does not tell much about what happens with the found blob, while `img_blob_exists` implies that the result will be boolean.